File: frontend/src/app/shared/service/admin/homework/HoweWorkQSTReponseAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {HoweWorkQSTReponseDto} from 'src/app/shared/model/homework/HoweWorkQSTReponse.model';
import {HoweWorkQSTReponseCriteria} from 'src/app/shared/criteria/homework/HoweWorkQSTReponseCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class HoweWorkQSTReponseAdminService extends AbstractService<HoweWorkQSTReponseDto, HoweWorkQSTReponseCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): HoweWorkQSTReponseDto {
        return new HoweWorkQSTReponseDto();
    }

    public constrcutCriteria(): HoweWorkQSTReponseCriteria {
        return new HoweWorkQSTReponseCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/howeWorkQSTReponse/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/homework/HoweWorkQSTReponseAdminService.java
package ma.zs.alc.service.facade.admin.homework;

import java.util.List;
import ma.zs.alc.bean.core.homework.HoweWorkQSTReponse;
import ma.zs.alc.dao.criteria.core.homework.HoweWorkQSTReponseCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface HoweWorkQSTReponseAdminService extends  IService<HoweWorkQSTReponse,HoweWorkQSTReponseCriteria>  {

    List<HoweWorkQSTReponse> findByHomeWorkQuestionId(Long id);
    int deleteByHomeWorkQuestionId(Long id);
    long countByHomeWorkQuestionRef(String ref);




}
